internal/commands/set_default_registry_test.go

```diff
@@ -0,0 +1,78 @@
+package commands_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/heroku/color"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	"github.com/buildpacks/pack/internal/commands"
+	"github.com/buildpacks/pack/internal/config"
+	ilogging "github.com/buildpacks/pack/internal/logging"
+	h "github.com/buildpacks/pack/testhelpers"
+)
+
+func TestSetDefaultRegistry(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+
+	spec.Run(t, "Commands", testSetDefaultRegistryCommand, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testSetDefaultRegistryCommand(t *testing.T, when spec.G, it spec.S) {
+	when("#SetDefaultRegistry", func() {
+		var (
+			outBuf     bytes.Buffer
+			logger     = ilogging.NewLogWithWriters(&outBuf, &outBuf)
+			tmpDir     string
+			configFile string
+			assert     = h.NewAssertionManager(t)
+		)
+
+		it.Before(func() {
+			var err error
+			tmpDir, err = ioutil.TempDir("", "pack-home-*")
+			assert.Nil(err)
+
+			configFile = filepath.Join(tmpDir, "config.toml")
+		})
+
+		it.After(func() {
+			_ = os.RemoveAll(tmpDir)
+		})
+
+		it("should set the default registry", func() {
+			cfg := config.Config{
+				Registries: []config.Registry{
+					{
+						Name: "myregistry",
+						URL:  "https://github.com/buildpacks/registry-index",
+						Type: "Github",
```

Comment:
Based on [this](https://github.com/buildpacks/rfcs/blob/main/text/0044-pack-publish-buildpack.md#type-github-schema), it should be `github`

```suggestion
						Type: "github",
```